modules/bar/workspaces/utils.ts

```diff
@@ -68,6 +68,70 @@ export const getWsColor = (
     return '';
 };
 
+export const getAppIcon = (
+    workspaceIndex: number,
+    removeDuplicateIcons: boolean,
+    {
+        iconMap,
+        defaultIcon,
+        emptyIcon,
+    }: {
+        iconMap: ApplicationIcons;
+        defaultIcon: string;
+        emptyIcon: string;
+    },
+) => {
+    // detect the clients attributes on the current workspace
+    const clients: ReadonlyArray<[className: string, title: string]> = hyprland.clients
```

Comment:
Nitpick: Prefer to keep type definitions in a type file.

Reply:
I have extracted the typings into `types/workspace.d.ts`. Please have a look at the naming if it fits your codebae.